__tests__/app.test.js

```diff
@@ -155,8 +154,152 @@ describe("app", () => {
+
+    test("It returns status code 201 and array with an object that have an updated body and author properties", () => {
+      const postartecle = [
+        {
+          username: "Waled",
+          body: "I have never expected this to be stressful.",
+        },
+      ];
+      const article_id = 3;
+      return request(app)
+        .post(`/api/articles/${article_id}/comments`)
+        .send(postartecle)
+        .expect(404)
+        .then(({body})=>{
+        expect(body.msg).toBe("User does not exist")
+
+    });
+    });
+  });
+
+  describe("8. PATCH /api/articles/:article_id", () => {
+    test("It Returns 200 status and an array of objects for a particular article", () => {
+      const article_id = 3;
+      const incremnt = { inc_votes: 100 };
+      return request(app)
+        .patch(`/api/articles/${article_id}`)
+        .send(incremnt)
+        .expect(200)
+        .then((result) => {
+         
+            expect(result.body[0]).toHaveProperty("article_id");
+            expect(result.body[0]).toHaveProperty("title");
+            expect(result.body[0]).toHaveProperty("topic");
+            expect(result.body[0]).toHaveProperty("author");
+            expect(result.body[0]).toHaveProperty("body");
+            expect(result.body[0]).toHaveProperty("created_at");
+            expect(result.body[0]).toHaveProperty("votes");
+            expect(result.body[0]).toHaveProperty("article_img_url");
+        });
+    });
+  
+    test("It Returns 200 status and an array of objects for a particular article with the updated votes property", () => {
+      const article_id = 1;
+      const incremnt = { inc_votes: 50 };
+      return request(app)
+        .patch(`/api/articles/${article_id}`)
+        .send(incremnt)
+        .expect(200)
+        .then((result) => {
+          expect(result.body[0].votes).toBe(150);
+        
+        });
+    });
+  
+    test("It Returns 200 status and an array of objects for a particular article with the updated votes property", () => {
+      const article_id = 9;
+      const incremnt = { inc_votes: -50 };
+      return request(app)
+        .patch(`/api/articles/${article_id}`)
+        .send(incremnt)
+        .expect(200)
+        .then((result) => {
+          expect(result.body[0].votes).toBe(-50);
+        
+        });
+    });
+  
+    test('Returns Error with 404 when the article id it does not exist', () => {
+      const article_id = 50;
+      return request(app)
+      .patch(`/api/articles/${article_id}`)
+      .expect(404)
+      .then(({body})=>{
+      expect(body.msg).toBe("Article not found")
+  
+        });
+      
+    });
+  
+    test("it returns a 400 status with a message : (Invalid id) if the endpoint is invalid", () => {
+      const article_id = "Waled";
+      return request(app)
+        .patch(`/api/articles/${article_id}`)
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Invalid id");
+        });
+    });
+  });
```

Comment:
You also need to add error handling for what the id is valid but doesnt exist i.e. 999, as well as tests for when incremnt isnt provided/has a missing inv_votes/isnt a number